libs/slam/async_slam/async_slam.cpp

```
@@ -377,8 +380,10 @@ void AsyncSlam::ProcessInput() {
       std::lock_guard image_guard(processing_images_mutex_);
       current_images = processing_images_;
     }
+    const auto current_image =
+        std::find_if(current_images.begin(), current_images.end(), [](const auto& image) { return image != nullptr; });
     const bool is_valid_image =
```

Clang (14.0.6)

[warning] variable 'is_valid_image' is not initialized

(cppcoreguidelines-init-variables)